klippa/src/glyf_loca.rs

```diff
@@ -0,0 +1,304 @@
+
+        if flags.contains(CompositeGlyphFlags::WE_HAVE_INSTRUCTIONS) {
+            we_have_instructions = true;
+        }
+
+        let old_gid = u16::from_be_bytes([out[i + 2], out[i + 3]]);
+        let Some(new_gid) = plan.glyph_map.get(&GlyphId::from(old_gid)) else {
+            return Vec::new();
+        };
+        let new_gid = new_gid.to_u32() as u16;
+        out[i + 2] = (new_gid >> 8) as u8;
+        out[i + 3] = (new_gid & 0xFF) as u8;
+
+        i += 4;
+
+        if flags.contains(CompositeGlyphFlags::ARG_1_AND_2_ARE_WORDS) {
+            i += 4;
+        } else {
+            i += 2;
+        }
+
+        if flags.contains(CompositeGlyphFlags::WE_HAVE_A_SCALE) {
+            i += 2;
+        } else if flags.contains(CompositeGlyphFlags::WE_HAVE_AN_X_AND_Y_SCALE) {
+            i += 4;
+        } else if flags.contains(CompositeGlyphFlags::WE_HAVE_A_TWO_BY_TWO) {
+            i += 8;
+        }
+
+        more = flags.contains(CompositeGlyphFlags::MORE_COMPONENTS);
+    }
+
+    if we_have_instructions {
+        if i + 1 >= len {
+            return Vec::new();
+        }
+        let instruction_len = u16::from_be_bytes([out[i], out[i + 1]]);
+        i += 2 + instruction_len as usize;
+    }
+
+    out.truncate(i);
+    out
+}
+
+fn trim_simple_glyph_padding(glyph_data: &[u8], num_coords: u16) -> usize {
+    let mut coord_bytes: usize = 0;
+    let mut coords_with_flags: u16 = 0;
+    let length = glyph_data.len();
+    let mut i: usize = 0;
+    while i < length {
+        let flag = SimpleGlyphFlags::from_bits_truncate(glyph_data[i]);
+        i += 1;
+
+        let mut repeat: u8 = 1;
+        if flag.contains(SimpleGlyphFlags::REPEAT_FLAG) {
+            if i >= length {
+                return 0;
+            }
+            repeat = glyph_data[i] + 1;
+            i += 1;
+        }
+
+        let mut x_bytes: u8 = 0;
+        let mut y_bytes: u8 = 0;
+        if flag.contains(SimpleGlyphFlags::X_SHORT_VECTOR) {
+            x_bytes = 1;
+        } else if !flag.contains(SimpleGlyphFlags::X_IS_SAME_OR_POSITIVE_X_SHORT_VECTOR) {
+            x_bytes = 2;
+        }
+
+        if flag.contains(SimpleGlyphFlags::Y_SHORT_VECTOR) {
+            y_bytes = 1;
+        } else if !flag.contains(SimpleGlyphFlags::Y_IS_SAME_OR_POSITIVE_Y_SHORT_VECTOR) {
+            y_bytes = 2;
+        }
+
+        coord_bytes += ((x_bytes + y_bytes) * repeat) as usize;
+        coords_with_flags += repeat as u16;
+        if coords_with_flags >= num_coords {
+            break;
+        }
+    }
+
+    if num_coords != coords_with_flags {
+        return 0;
+    }
+    i += coord_bytes;
+    i
+}
+
+fn subset_head(head: &Head, loca_format: u8) -> Result<Vec<u8>, SubsetError> {
+    let mut out = Vec::new();
+    out.extend_from_slice(head.offset_data().as_bytes());
+
+    let Some(index_loca_format) = out.get_mut(50..52) else {
+        return Err(SubsetTableError(Head::TAG));
+    };
```

Comment:
if we have a `Head` table at this point it mens that we've already validated that it is the correct length, so this branch is unreachable.

Reply:
and given that, this can all be written as,

`out.get_mut(50..52).unwrap().copy_from_slice(&[0, loca_format]);`